cirq-core/cirq/protocols/decompose_protocol.py

```diff
@@ -138,6 +148,9 @@ def decompose(
         None, Exception, Callable[['cirq.Operation'], Optional[Exception]]
     ] = _value_error_describing_bad_operation,
     preserve_structure: bool = False,
+    qubit_manager: Optional['cirq.QubitManager'] = None,
+    constraint: DecomposeConstraints = lambda _: True,
```

Comment:
Is `constraint` doing same job as `keep` argument above?  What's the difference, if any?

Reply:
constraint doesn't mean we want to keep the operation ... but that we want to either keep it or decompose it.

For exmple a constraint might be a check that the operation either belongs to a gateset or can be decomposed in terms of that gateset

Reply:
removed constraints